test/flops.h

```diff
@@ -103,11 +103,29 @@ static double  flops_sgemm(double m, double n, double k)
     { return    fmuls_gemm(m, n, k) +    fadds_gemm(m, n, k); }
 
 //------------------------------------------------------------ gbmm
+// usually the bottom equation returned calculates the flops,
+// but some matrices are too long or too wide and require extra care
+// think rectangle minus trapezoid minus triangle and reduce:
+//        (m*k - (m-kl+m-k-kl-1.0)*0.5*k - (k-ku-1.0)*(k-ku)*0.5)*n;
+//        (m*k - (m-kl)*k+(k-1.0)*k*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
+//        (kl*k + (k-1.0)*k*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
 static double fmuls_gbmm(double m, double n, double k, double kl, double ku)
-    { return (m * k - (m - kl - 1.0) * (m - kl) * 0.5 - (k - ku - 1.0) * (k - ku) * 0.5) * n; }
+{ 
+    if(m > k+kl)
+        return (kl*k + (k-1.0)*k*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
+    if(k > m+ku)
+        return (ku*m - (m-kl-1.0)*(m-kl)*0.5 + (m-1.0)*m*0.5)*n;
+    return (m*k - (m-kl-1.0)*(m-kl)*0.5 - (k-ku-1.0)*(k-ku)*0.5)*n;
+}
 
 static double fadds_gbmm(double m, double n, double k, double kl, double ku)
-    { return (m * k - (m - kl - 1.0) * (m - kl) * 0.5 - (k - ku - 1.0) * (k - ku) * 0.5) * n; }
+{ 
```

Comment:
We don't do spaces at the end of line.

Reply:
I didn't even notice this space was present. My apologies - I will remove it!